datamade/just-spaces
# Data handler

## Overview

This PR implements a custom Fobi data handler to store collected survey data in PLDP-described models. 

### Notes

See comments in `fobi_form_handlers.py` for details on the parts of this that will need more work once we address #63. Study, Location, and Total fields are handled with static placeholder values for now.

Something to consider in `django-pldp`: now that SurveyComponent is more generic, it no longer makes sense to have a detail_level field that defaults to "basic". Maybe we should switch that to None?

Also, this needs some tests! @reginafcompton has kindly agreed to help me get started on that in the next few days, so I can pick it up on Monday.

## Testing Instructions

* Make sure to update requirements and pull in new migrations! Several things have changed in `django-pldp`
* Create a new survey locally. Check to make sure it has "Collect data" as a handler added automatically
<img width="991" alt="Screen Shot 2019-03-27 at 4 58 36 PM" src="https://user-images.githubusercontent.com/1094243/55115085-9b830800-50b1-11e9-8c4b-d26c9b0c167b.png">

* Add at least one sample question to your new survey
* Run the survey! Submit an answer to your question
* Check the database. Do you see what you just submitted in the `pldp_survey`, `pldp_surveyrow`, and `pldp_surveycomponent` tables?

Connects #25
